### PR TITLE
fix: geo 657 validate end date is always before today

### PR DESCRIPTION
**Pull request by goemen**
# Description

Validate end date is always before today

Fixes # ([issue](https://finrms.atlassian.net/browse/GEO-657))

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://pay-transparency-pr-513-frontend.apps.silver.devops.gov.bc.ca)
- [Admin Frontend](https://pay-transparency-pr-513-admin-frontend.apps.silver.devops.gov.bc.ca)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/fin-pay-transparency/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/fin-pay-transparency/actions/workflows/merge.yml)

### CODE REVIEW

File: backend/src/v1/services/external-consumer-service.ts [hunk 1]
```diff
@@ -11,14 +11,22 @@ import { Prisma } from '@prisma/client';
...
    * This function returns a list of objects with pagination details to support the analytics team.
-   * this endpoint should not return more than 1000 records at a time.
-   * if limit is greater than 1000, it will default to 1000.
-   * calling this endpoint with no limit will default to 1000.
+   * this endpoint should not return more than 50 records at a time.
+   * if limit is greater than 50, it will default to 50.
+   * calling this endpoint with no limit will default to 50.
    * calling this endpoint with no offset will default to 0.
-   * calling this endpoint with no start date will default to 30 days ago.
+   * calling this endpoint with no start date will default to yesterday.
```
Comment (sukanya-rath) [inline backend/src/v1/services/external-consumer-service.ts:29]: are we not supposed to work at -1 days?

File: backend/src/v1/services/external-consumer-service.ts [hunk 2]
```diff
@@ -32,35 +40,31 @@ const externalConsumerService = {
...
 
       if (endDate) {
         const date = convert(LocalDate.parse(endDate)).toDate();
-        endDt = LocalDateTime.from(nativeJs(date, ZoneId.UTC))
-          .withHour(23)
-          .withMinute(59);
+        endDt = withEndOfDay(LocalDateTime.from(nativeJs(date, ZoneId.UTC)));
       }
     } catch (error) {
+      console.log(error);
```
Comment (sukanya-rath) [inline backend/src/v1/services/external-consumer-service.ts:67]: remove this

File: backend/src/v1/services/external-consumer-service.ts (hunk 1 above)
Comment (sukanya-rath) [reply backend/src/v1/services/external-consumer-service.ts:29]: add to integration test scenarios

**Review by sukanya-rath — CHANGES_REQUESTED** [review]

File: backend/src/v1/services/external-consumer-service.ts (hunk 1 above)
Comment (goemen) [reply backend/src/v1/services/external-consumer-service.ts:29]: Yes, I was manually testing this last night, and I noticed diff results between the unit tests and swagger results

**Review by sukanya-rath — APPROVED** [review]

**Pull request merged**